src/smt/theory_str_noodler/theory_str_noodler_final_check.cpp

```diff
@@ -266,9 +281,26 @@ namespace smt::noodler {
             result = dec_proc->compute_next_solution();
             if (result == l_true) {
                 auto [noodler_lengths, precision] = dec_proc->get_lengths();
-                lengths = len_node_to_z3_formula(noodler_lengths);
+
+                LenFormulaContext context = {
+                    .manager = this->m,
+                    .arith_utilities = this->m_util_a,
+                    .seq_utilities = this->m_util_s,
+                    .quantified_vars = this->quantif_vars,
+                    .known_z3_exprs = this->var_name
+                };
+
+                lengths = convert_len_node_to_z3_formula(context, noodler_lengths);
+
+                if (0) {
+                    std::ofstream out_file("./not-contains-lia.smt2");
+                    write_z3_expr_into_stream(this->m, out_file, lengths);
+                    // write_len_formula_as_smt2(noodler_lengths, out_file);
+                    out_file.close();
+                }
```

Comment:
Remove this, or add it as a trace, like
```suggestion
                STRACE("str-print-lia",
                    std::ofstream out_file("./not-contains-lia.smt2");
                    write_z3_expr_into_stream(this->m, out_file, lengths);
                    // write_len_formula_as_smt2(noodler_lengths, out_file);
                    out_file.close();
                );
```

Reply:
I've added a strace call, like you have suggested. Thank you.